src/Components/Components/test/RenderTreeDiffBuilderTest.cs

```diff
@@ -1257,6 +1257,123 @@ public void AttributeDiff_WithSequentialSequenceNumber_AttributeRemovedAtEnd()
                 });
         }
 
+        [Fact]
+        public void AttributeDiff_WithClosure_SameValue()
+        {
+            // Arrange
+            var i = 1;
+
+            void BuildRenderTree(RenderTreeBuilder builder)
+            {
+                var j = i;
+                builder.OpenElement(0, "My element");
+                builder.AddAttribute(1, "attr1", (Action)(() => Console.WriteLine($"{j}")));
+                builder.CloseElement();
+            }
+
+            BuildRenderTree(oldTree);
+            BuildRenderTree(newTree);
+
+            // Act
+            var (result, referenceFrames) = GetSingleUpdatedComponent();
+
+            // Assert
+            Assert.Empty(result.Edits);
+        }
+
+        [Fact]
+        public void AttributeDiff_WithClosure_DifferentValue()
+        {
+            // Arrange
+            var i = 1;
+
+            void BuildRenderTree(RenderTreeBuilder builder)
+            {
+                var j = i;
+                builder.OpenElement(0, "My element");
+                builder.AddAttribute(1, "attr1", (Action)(() => Console.WriteLine($"{j}")));
+                builder.CloseElement();
+            }
+
+            BuildRenderTree(oldTree);
+            i++;
+            BuildRenderTree(newTree);
+
+            // Act
+            var (result, referenceFrames) = GetSingleUpdatedComponent();
+
+            // Assert
+            Assert.Collection(
+                result.Edits,
+                entry =>
+                {
+                    AssertEdit(entry, RenderTreeEditType.SetAttribute, 0);
+                });
```

Comment:
```suggestion
                entry =>AssertEdit(entry, RenderTreeEditType.SetAttribute, 0));
```

Also in equivalent lines below